### ISSUE TITLE
iOS9 saving bug.

### ISSUE THREAD

**Issue by ChrisOSX** [Kjuly/ALAssetsLibrary-CustomPhotoAlbum#45]
I've been using this library for a while and since I got it working correctly I haven't had any issues.

However, now I'm noticing on iOS 9 (now running WKWebView instead of UIWebView, not sure if that makes a difference), that when I save, it saves to my custom album, camera roll (not a problem as that's supposed to happen), but it saves to a Screenshot folder as well.

Tested in iOS 8 simulator, no screenshot folder.

Is this a iOS 9 issue? Library issue? WKWebView issue?

Thanks for any feedback.

**Comment by Kjuly** [comment]
Hi @ChrisOSX this lib won't try to take a screenshot, it uses system SDK to take photo & save it custom album.

I just tried in iOS 9.2 (13C75), w/o anything wrong. :)

**Comment by ChrisOSX** [comment]
Just took these 2 simulator screenshots of iOS 8.4 and 9.2 to show the comparison. 

![simulator screen shot ios 8 4](https://cloud.githubusercontent.com/assets/4047356/12261461/4742352a-b8ef-11e5-8aaa-4481c164a5fe.png)
![simulator screen shot ios 9 2](https://cloud.githubusercontent.com/assets/4047356/12261515/959e40a6-b8ef-11e5-9833-d9b896c71153.png)

**Comment by Kjuly** [comment]
Can u offer the code snippet u used to save the photo?

**Comment by ChrisOSX** [comment]
This is how I'm calling it in my file.
```
        NSURL *imageURL = self.imageInfo.imageURL;
        UIImage *image = [UIImage imageWithData:[NSData dataWithContentsOfURL:imageURL]];
        
        // The completion block to be executed after image taking action process done
        void (^completion)(NSURL *, NSError *) = ^(NSURL *assetURL, NSError *error) {
            if (error) NSLog(@"!!!ERROR,  write the image data to the assets library (camera roll): %@",
                             [error description]);
            NSLog(@"*** URL %@ | %@ || type: %@ ***", assetURL, image, [assetURL class]);
            
        };
        
        void (^failure)(NSError *) = ^(NSError *error) {
            if (error == nil) return;
            NSLog(@"!!!ERROR, failed to add the asset to the custom photo album: %@", [error description]);
        };
        
        [self.assetsLibrary saveImage:image
                              toAlbum:kKYCustomPhotoAlbumName
                           completion:completion
                              failure:failure];
```

**Comment by Kjuly** [comment]
What about the code snippet to generate `self.imageInfo`?

**Comment by ChrisOSX** [comment]
I'm using JTSImageViewController. It has quite a few files.

https://github.com/jaredsinclair/JTSImageViewController

**Comment by Kjuly** [comment]
Not sure about it, just checked its demo, seems okay. Can u offer the code snippet u used to generate `self.imageInfo`? Any other lib u used related to image taking?

**Comment by ChrisOSX** [comment]
self.imageInfo is generated by that library. It's it own .h and .m files then linked to the other library files.

I don't use any other lib for image taking, other than this one.

My action sheet is triggered by a long press gesture, and one button is for saving, where this library comes into play.

**Comment by Kjuly** [comment]
Seems this part is ok, JTSImageInfo instance just stores image related data. No idea then, sorry, it's hard for me to debug w/o a testing project, i'd glad to offer more help if u can give me a simple demo project that can reproduce the issue (just upload it as ur own repo).

I suggest u do not use two libs (or another other) at same time first, and test to c which one generates the screenshot actually. e.g., don't use ALAssetsLibrary-CustomPhotoAlbum lib, just save it to normal album to c whether the screenshot still appear.

Btw, have u tried this lib's demo, does the screenshot appear as well in ur device?

Let me know if u're pretty sure it's indeed lead by this lib, thx :)

**Comment by ChrisOSX** [comment]
So with some troubleshooting and using an older version of my project (which uses a different image viewer), I can determine that the new image viewer isn't really to blame, but how I loaded the image and calling the save action. I renamed some things in the save function. did a full clean and such and then tested it out on my device. No screenshot folder.

I had a feeling it was that, but none the less I appreciate the quick feedback and help. This is a awesome library!
